trl/trainer/online_dpo_trainer.py

```diff
@@ -505,11 +503,11 @@ def repeat_generator():
                 metrics["lr"] = self.lr_scheduler.get_last_lr()[0]
                 metrics["episode"] = self.state.episode
                 self.state.epoch = self.state.episode / self.train_dataset_len  # used by self.log
-                self.state.global_step += 1
```

Comment:
Moved this to occur after the LR scheduler update, similar to how `transformers.Trainer` does it.